File: TravelPlannerApp/TravelPlanService/Models/TravelPlan.cs
namespace TravelPlanService.Models
{
    public class TravelPlan
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public decimal Budget { get; set; }
        public string Notes { get; set; } = string.Empty;
        public int UserId { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public List<Destination> Destinations { get; set; } = new();
        public List<Activity> Activities { get; set; } = new();
    }
}

File: frontend/src/models/TravelPlan.js
export const createTravelPlan = (data) => ({
  id:           data.id           || null,
  name:         data.name         || '',
  description:  data.description  || '',
  startDate:    data.startDate    || '',
  endDate:      data.endDate      || '',
  budget:       data.budget       || 0,
  notes:        data.notes        || '',
  userId:       data.userId       || null,
  createdAt:    data.createdAt    || '',
  destinations: data.destinations || [],
  activities:   data.activities   || [],
});
